src/sentry/static/sentry/app/utils/discover/eventView.tsx

```diff
@@ -388,6 +388,55 @@ class EventView {
     });
   }
 
+  static fromSavedQueryOrLocation(
+    saved: SavedQuery | undefined,
+    location: Location
+  ): EventView {
+    let fields = decodeFields(location);
+    const {start, end, statsPeriod} = location.query;
+    const id = decodeScalar(location.query.id);
+    const projects = decodeProjects(location);
+    const sorts = decodeSorts(location);
+    const environments = collectQueryStringByKey(location.query, 'environment');
+
+    if (saved) {
+      if (fields.length === 0) {
+        fields = saved.fields.map((field, i) => {
+          const width =
+            saved.widths && saved.widths[i]
+              ? Number(saved.widths[i])
+              : COL_WIDTH_UNDEFINED;
+
+          return {field, width};
+        });
+      }
+      const {savedStart, savedEnd, savedStatsPeriod} = getParams({
+        start: saved.start,
+        end: saved.end,
+        statsPeriod: saved.range,
+      });
+      return new EventView({
+        id: id || saved.id,
+        name: decodeScalar(location.query.name) || saved.name,
+        fields,
+        query: decodeQuery(location) || queryStringFromSavedQuery(saved),
+        project: projects,
```

Comment:
Project and environment can be set/modified on global selection header and I want that to propagate that down to the event view. so I'm going to use the location params to set these props instead of the saved query